allennlp/modules/seq2seq_encoders/stacked_self_attention.py

```diff
@@ -0,0 +1,159 @@
+    architecture in `Attention is all you Need
+    <https://www.semanticscholar.org/paper/Attention-Is-All-You-Need-Vaswani-Shazeer/0737da0767d77606169cbf4187b83e1ab62f6077>`_ .
+
+    This encoder combines 3 layers in a 'block':
+    1. A 2 layer FeedForward network.
+    2. Multi-headed self attention, which uses 2 learnt linear projections
+       to perform a dot-product similarity between every pair of elements
+       scaled by the square root of the sequence length.
+    3. Layer Normalisation.
+
+    These are then stacked into ``num_layers`` layers.
+
+    Parameters
+    ----------
+    input_dim : ``int``, required.
+        The input dimension of the encoder.
+    hidden_dim : ``int``, required.
+        The hidden dimension used for the self attention layers.
+    projection_dim : ``int``, required.
+        The dimension of the linear projections for the self-attention layers.
+    non_linear_hidden_dim : ``int``, required.
+        The middle dimension of the FeedForward network. The input and output
+        dimensions are fixed to ensure sizes match up for the self attention layers.
+    num_layers : ``int``, required.
+        The number of stacked self attention -> feedfoward -> layer normalisation blocks.
+    num_attention_heads : ``int``, required.
+        The number of attention heads to use per layer.
+    internal_similarity : ``SimilarityFunction``, optional (default = DotProductSimilarity(scale_output=True)).
+        The internal similarity function to use for creating the attention distributions per
+        attention head.
+    use_positional_encoding: ``bool``, optional, (default = True)
+        Whether to add sinusoidal frequencies to the input tensor. This is strongly recommended,
+        as without this feature, the self attention layers have no idea of absolute or relative
+        position (as they are just computing pairwise similarity between vectors of elements),
+        which can be important features for many tasks.
+    """
+
+    def __init__(self,
+                 input_dim: int,
+                 hidden_dim: int,
+                 projection_dim: int,
+                 non_linear_hidden_dim: int,
+                 num_layers: int,
+                 num_attention_heads: int,
+                 internal_similarity: SimilarityFunction = DotProductSimilarity(scale_output=True),
+                 use_positional_encoding: bool = True,
+                 dropout_prob: float = 0.2) -> None:
+        super(StackedSelfAttentionEncoder, self).__init__()
+
+        self._use_positional_encoding = use_positional_encoding
+        self._attention_layers: List[IntraSentenceAttentionEncoder] = []
+        self._feedfoward_layers: List[FeedForward] = []
+        self._layer_norm_layers: List[LayerNorm] = []
+
+        feedfoward_input_dim = input_dim
+        for i in range(num_layers):
+            # Project output of attention encoder through a feedforward network
```

Comment:
The way this is worded it makes me think you're doing the operation here, but it actually happens in `forward`.